### PR TITLE
feat: enable mdformat through treefmt-nix

### PR DESCRIPTION
**Pull request by o-az**
adds markdown formatting

mdformat: https://mdformat.readthedocs.io/en/stable/index.html

treefmt-nix options: https://flake.parts/options/treefmt-nix#opt-perSystem.treefmt.programs.mdformat

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ For more complex conversations, use the [discussions](https://github.com/unionla
 
 We evaluate the need for a PR based on:
 
-1. Severity of the issue (bug or feature request).
-2. Maintainability: will this become a burden for little gain, or add value?
-3. Can the core team understand the code additions being made, and maintain them, or will they rely on you in the future?
+1. Severity of the issue (bug or feature request),
+1. Maintainability: will this become a burden for little gain, or add value?
+1. Can the core team understand the code additions being made, and maintain them, or will they rely on you in the future?
```
Comment (benluelo) [inline CONTRIBUTING.md:17]: what lol

File: flake.lock [hunk 2]
```diff
@@ -486,11 +486,11 @@
...
     "nixpkgs-unstable": {
       "locked": {
-        "lastModified": 1724479785,
-        "narHash": "sha256-pP3Azj5d6M5nmG68Fu4JqZmdGt4S4vqI5f8te+E/FTw=",
+        "lastModified": 1725103162,
+        "narHash": "sha256-Ym04C5+qovuQDYL/rKWSR+WESseQBbNAe5DsXNx5trY=",
         "owner": "NixOS",
         "repo": "nixpkgs",
-        "rev": "d0e1602ddde669d5beb01aec49d71a51937ed7be",
+        "rev": "12228ff1752d7b7624a54e9c1af4b222b3c1073b",
```
Comment (benluelo) [inline flake.lock:493]: we're sure we can safely update nixpkgs-unstable?

File: lib/ics23/README.md [hunk 3]
```diff
@@ -4,86 +4,86 @@ Below an overview of all data types from the [ICS-23 specification](https://gith
 
 ## CommitmentState
 
-spec: `object`  
+spec: `object`\
 impl: _not implemented_
```
Comment (benluelo) [inline lib/ics23/README.md:8]: what does this do?

File: CONTRIBUTING.md (hunk 1 above)
Comment (o-az) [reply CONTRIBUTING.md:17]: yes

Comment (benluelo) [reply CONTRIBUTING.md:17]: the more you know

File: lib/ics23/README.md (hunk 3 above)
Comment (o-az) [reply lib/ics23/README.md:8]: The `\` at the end of the line is a Markdown line break.

It does two things:

1. Ensures a line break after this line
2. Prevents unintended merging with the next line

This is useful for:
- Maintaining specific formatting
- Separating content that shouldn't be combined
- Creating precise line breaks without starting a new paragraph

Without it, some Markdown renderers might combine adjacent lines.
___
in other words, it denotes a hardline break
https://mdformat.readthedocs.io/en/stable/users/style.html#hard-line-breaks
(`mdformat` applied it automatically)

File: flake.lock (hunk 2 above)
Comment (o-az) [reply flake.lock:493]: that's the guidance I got from @cor - to update it and if there are any issues to raise and try to fix. This is what I did last few times (last update 5 days ago https://github.com/unionlabs/union/commit/7e38400a540f7f1c5d11f628cd9caa7bf4d0fca3)

File: CONTRIBUTING.md (hunk 1 above)
Comment (cor) [reply CONTRIBUTING.md:17]: I'm in doubt regarding this. I mostly rawdog .md files in the terminal rather than looking at the rendered output

Comment (cor) [reply CONTRIBUTING.md:17]: On the other hand, it does make diffs much nicer

**Review by cor — APPROVED** [review]
